Mathlib/Combinatorics/SimpleGraph/Path.lean

```diff
@@ -1053,6 +1065,70 @@ lemma iUnion_connectedComponentSupp (G : SimpleGraph V) :
   simp only [Set.mem_range, SetLike.mem_coe]
   exact ⟨by use G.connectedComponentMk v; exact rfl, rfl⟩
 
+namespace ConnectedComponent
+
+/-- A set of vertices represents a set of components if it contains exactly
+one vertex from each component.
+-/
+structure _root_.Set.Represents (s : Set V) (C : Set (G.ConnectedComponent)) where
```

Comment:
Hmm, looking at this definition, it doesn't really seem like it's specific to graphs... That is, given a partition of a subset, it's reasonable to define a set of representatives for it. I could, however, believe that this formulation (using connectedComponent rather than an arbitrary partition) is helpful in your application. But it certainly doesn't look like it belongs in SimpleGraph/Path. Can we move this to a subfolder on connected components instead?

Reply:
I have done this, but I have also put together #21097 as an alternative. In the past I have made missteps with Quot, and I'm still not really certain on the nuances and design of the Quot/ConnectedComponent situation.

If the alternative makes sense, I wonder if it also makes sense to migrate off supp and use the SetLike instance instead, so feel free to weigh in if you have any thoughts on that:)

Reply:
I just noticed we have the SetLike instance I introduced in that PR already for [connected components](https://github.com/leanprover-community/mathlib4/blob/413249c638b4c9453166d2b94c9a7f13f6238b14/Mathlib/Combinatorics/SimpleGraph/Path.lean#L1065)

Is there a particular reason we use .supp explicitly and tend not to rely on this instance?